pkg/utils/gardener/shootstate/shootstate.go

```diff
@@ -61,3 +66,83 @@ func computeGardenerData(
 
 	return dataList, nil
 }
+
+func computeExtensionsDataAndResources(
+	ctx context.Context,
+	seedClient client.Client,
+	seedNamespace string,
+) (
+	[]gardencorev1beta1.ExtensionResourceState,
+	[]gardencorev1beta1.ResourceData,
+	error,
+) {
+	var (
+		dataList  []gardencorev1beta1.ExtensionResourceState
+		resources []gardencorev1beta1.ResourceData
+	)
+
+	for _, extension := range []struct {
+		objKind           string
+		newObjectListFunc func() client.ObjectList
+	}{
```

Comment:
You could use a `map[string]client.ObjectList` instead

Reply:
I was using this before but intentionally switched to the struct so that the order in `.spec.{extensions,resources}` doesn't change on every execution.

Reply:
Alright, got it